library/core/src/ascii/ascii_char.rs

```diff
@@ -449,6 +453,7 @@ impl AsciiChar {
     /// or returns `None` if it's too large.
     #[unstable(feature = "ascii_char", issue = "110998")]
     #[inline]
+    #[ensures(|result| (b <= 127) == (result.is_some() && result.unwrap() as u8 == b))]
```

Comment:
It occurred to me that it's slightly cleaner to write this as:
```suggestion
    #[ensures(|result| (b <= 127) == (result = Some(x) && x as u8 == b))]
```
but I think this requires `#![feature(let_chains)]` which is still unstable.

Reply:
Can you do `result = Some(x)` without an initial `if let` or `while let` expression? I thought the idea of the let chaining feature was to chain additional `let`s like this only after a leading `if let` or `while let` expression.
I could write this instead:
`(if let Some(x) = result { *x as u8 == b } else {false} )`
but IMO that's more confusing than what we have now.

Reply:
You're right, that's not possible.

Seems the way to do this is to use a `matches!`, e.g.
```rust
matches!(result, Some(x) if x as u8 == b)
```
```